scripts/export_large_tsv/export_large_tsv.py

```diff
@@ -0,0 +1,117 @@
+    entity_types_json = response.json()
+    entity_count = entity_types_json[entity]["count"]
+    attribute_names = entity_types_json[entity]["attributeNames"]
+    entity_id = entity_types_json[entity]["idName"]
+    # add the entity_id value to list of attributes (not a default attribute of API response)
+    attribute_names.insert(0, entity_id)
+
+    print(f'{entity_count} {entity}(s) to export.')
+
+    with open(tsv_name, "w") as tsvout:
+        # add header with attribute values to tsv
+        tsvout.write("\t".join(attribute_names) + "\n")
+        # set starting row value and calculate number of pages
+        row_num = 0
+        num_pages = int(math.ceil(float(entity_count) / page_size))
+
+        # get entities by page where each page has page_size # of rows using API call
+        print(f'Getting all {num_pages} pages of entity data.')
+        all_page_responses = []
+        for page in tqdm(range(1, num_pages + 1)):
+            all_page_responses.append(get_entity_by_page(project, workspace, entity, page, page_size))
+
+        # for each response(page) in all_page_responses[] - contains parameter metadata
+        print(f'Writing {entity_count} attributes to tsv file.')
+        for page_response in tqdm(all_page_responses):
+            # for each set of attributes in results (no parameters) get attribute names and entity_id(name)
+            for entity_json in page_response["results"]:
+                attributes = entity_json["attributes"]
+                name = entity_json["name"]
+                # add name and value to dictionary of attributes
+                attributes[entity_id] = name
+
+                values = []
+                # for each attribute(column name) in list of attribute names(all columns for entity)
+                for attribute_name in attribute_names:
+                    value = ""
+                    # if entity's attribute(column) is in list of attributes from response, set response's attribute value
+                    if attribute_name in attributes:
+                        value = attributes[attribute_name]
+
+                    values.append(value)
+
+                tsvout.write("\t".join(values) + "\n")
+                row_num += 1
+
+    print(f'Finished exporting {entity}(s) to tsv with name {tsv_name}.')
+
+
+if __name__ == "__main__":
+    # argument parser
+    parser = argparse.ArgumentParser(description="Exports/downloadload a TSV file from Terra when it is too large to download via the UI.")
+    # application arguments
+    parser.add_argument(
+        '--project',
+        type=str,
+        action='store',
+        required=True,
+        help='Terra namespace/project of workspace.')
+    parser.add_argument(
+        '--workspace',
+        type=str,
+        action='store',
+        required=True,
+        help='Name of Terra workspace.')
+    parser.add_argument(
+        '--entity',
```

Comment:
i'd change this to `entity_type` just for full clarity and consistency

Reply:
good call, i changed that actually in the script as well just to be super clear and also readme.